### ISSUE TITLE
bun not running local package bin

### ISSUE THREAD

**Issue by jdion84** [oven-sh/bun#5075]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

create file `my_modules/bunclitest/package.json`:

```js
{
  "name": "bunclitest",
  "module": "index.ts",
  "type": "module",
  "bin": {
    "bunclitest": "bin.js"
  }
}
```

create file `my_modules/bunclitest/bin.js`:

```js
#!/usr/bin/env node

console.log("hello bin");
```

open `my_modules` dir in terminal and run `bun link`

open project root folder and run `bun link bunclitest --save`

run `bunx myclitest`

nothing is outputted to console.

### What is the expected behavior?

`hello bin` should be outputted to console when running `bunx bunclitest`

### What do you see instead?

nothing

**Comment by cyfung1031** [comment]
Same issue.
From my case, `error: missing script XXXXX` is printed.

I believe this is MacOS specific bug.

**Comment by futured-it** [comment]
Yeah. Same thing, on Mac too. I guess, `bun link ...` simply doesn't register your project to bash commands as Node does.

**Comment by futured-it** [comment]
I guess, as a temporary solution you could link it with Node.js using `npm link`, and add `#!/usr/bin/env bun` inside of your JS (or TS) runtime file. That's stupid, but probably it'll work.

**Comment by futured-it** [comment]
I did some researches and realised that the fix is so easy:
You should use string as a `bin` value, not object, so instead of this:
```json
{
  "name": "bunclitest",
  "module": "index.ts",
  "type": "module",
  "bin": {
    "bunclitest": "bin.js"
  }
}
```
do like this:
```json
{
  "name": "bunclitest",
  "module": "index.ts",
  "type": "module",
  "bin": "bin.js"
}
```

**Comment by jdion84** [comment]
> You should use string as a bin value, not object

no, you shouldnt

**Comment by futured-it** [comment]
> > You should use string as a bin value, not object
> 
> no, you shouldnt

Well, in my situation that's fixed the problem. It doesn't work with object on my and these peoples machines for some reasons.

**Comment by cpakken** [comment]
bin commands still not working with bun linked packages as of version 1.18

**Comment by kejedi** [comment]
I figured this out

after you add the bin to the local package, just do `bun link` on it again

**Comment by mkv27** [comment]
If you install bun via homebrew and do nothing else is not going to work. Installing via `curl -fsSL https://bun.sh/install | bash` solve the problem

**Comment by kevbot-git** [comment]
> If you install bun via homebrew and do nothing else is not going to work. Installing via `curl -fsSL https://bun.sh/install | bash` solve the problem

This worked for me too 🎉

**Comment by jonmagic** [comment]
I believe bun links to a folder like `~/.bun/bin` or the equivalent for your operating system. On MacOS after installing bun via homebrew I was able to fix the issue by updating my PATH to include `$HOME/.bun/bin`.

**Comment by jacobdalamb** [comment]
Can this be addressed soon please! Using 'bun link` is useless without local package bin created within local package installation.
